api/read/read.go

```diff
@@ -0,0 +1,31 @@
+// Package read implements request object readers.
+package read
+
+import (
+	"encoding/json"
+	"io"
+
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smallstep/certificates/errs"
+)
+
+// JSON reads JSON from the request body and stores it in the value
+// pointed by v.
+func JSON(r io.Reader, v interface{}) error {
+	if err := json.NewDecoder(r).Decode(v); err != nil {
+		return errs.BadRequestErr(err, "error decoding json")
```

Comment:
If this is always used with the request body, it's probably better to show
```suggestion
		return errs.BadRequestErr(err, "error reading request body")
```

Reply:
I, too, think this should be changed but it'd break the test suite of other packages. For now, I'll leave it as it is!